lib/ddtrace/contrib/resque/resque_job.rb

```diff
@@ -0,0 +1,30 @@
+require 'ddtrace/ext/app_types'
+
+module Datadog
+  module Contrib
+    # Uses Resque job hooks to create traces
+    module ResqueJob
+      SERVICE = 'resque'.freeze
+
+      def self.extended(base)
+        Datadog::Pin.new(SERVICE, app_type: Ext::AppTypes::WORKER).onto(base)
+      end
+
+      # rubocop:disable Style/RedundantSelf
+      def around_perform(*args)
+        pin = self.datadog_pin
+        pin.tracer.set_service_info(SERVICE, 'resque', Ext::AppTypes::WORKER)
```

Comment:
we should not do this in the `around_perform`, because it means making this calls before each execution. This call must be done only once per initialization, so it would be great if we can find another hook or class initialization to send the service metadata.